### PR TITLE
Implementing Date and Time pickers

### PR DESCRIPTION
**Pull request by grzesiek2010**

### CODE REVIEW

**Comment by grzesiek2010** [comment]
@lognaturel please look at this when you are free. I've implemented pickers according to your suggestion. Gennerally it looks and works well but there is one problem - DateWidget (or DateTime) doesn't look good when we use "calendar" appearance as there is no enough space. 
This approach is deprecated anyway so I think we should wait for https://github.com/opendatakit/collect/issues/464 and then use view like: https://i.stack.imgur.com/gsPyb.png if we want to keep calendar.

I've attahced a form and thanks to that you can test all widgets:
[dates.txt](https://github.com/opendatakit/collect/files/878646/dates.txt)

**Comment by lognaturel** [comment]
Cool! That's a big improvement. For this issue, we should just focus on the spinners and not worry about the calendars. I agree with you that it probably makes most sense to do as part of the broader UI refresh.

I believe that it should be possible to have a full date that sets day month and year with a `no-calendar` appearance. Same with datetime, it should be possible to set the date with spinners and then the time.

DateYearMonth and DateYear work well but the dialog title includes a full date with day, month, year which is confusing. Is it possible to either limit it only to the date components that are available for selection or replace it by some fixed string like "Select year"?

**Comment by lognaturel** [comment]
Also, good commit messages. 👍

**Comment by grzesiek2010** [comment]
I wasn't able to set the dialog title permanently or in [this method](https://github.com/opendatakit/collect/pull/789/files#diff-71ee66035a9814fa3226aa0ab0ddfd23R191) it doesn't work so I added a small trick removing the title at all.

**Comment by grzesiek2010** [comment]
@lognaturel new form for testing: 
[dates.txt](https://github.com/opendatakit/collect/files/906091/dates.txt)

**Comment by lognaturel** [comment]
For setting the dialog title, did you try creating your own subclasses of `DatePickerDialog` and overriding `getTitle`? You should be able to create `DatePickerDialogYearOnly` and `DatePickerDialogMonthOnly` or something like that. [Here](http://stackoverflow.com/a/20772570/137744) is more.

File: collect_app/src/main/res/values/untranslated.xml [hunk 1]
```diff
@@ -15,5 +15,8 @@
     <string translatable="false" name="default_server_url">https://opendatakit.appspot.com</string>
 	<string translatable="false" name="default_odk_formlist">/formList</string>
 	<string translatable="false" name="default_odk_submission">/submission</string>
-    <item name="mapview" type="id"/>
+	<string translatable="false" name="time">%1$s:%2$s</string>
```
Comment (lognaturel) [inline collect_app/src/main/res/values/untranslated.xml]: This doesn't seem right, the order should be different based on locale.

**Comment by lognaturel** [comment]
Ah, I just realized that this removes the calendar-based date selection altogether. That wasn't the intent. The intent was to leave the calendar-based implementation as it was and improve the spinner-based one. Let's talk on Slack when we're both available and see what makes the most sense to do.

**Comment by grzesiek2010** [comment]
@lognaturel 
I've fixed calendar and the problem with dialog title.

File: collect_app/src/main/java/org/odk/collect/android/widgets/DateWidget.java [hunk 2]
```diff
@@ -225,17 +132,114 @@ public void setFocus(Context context) {
...
+    private void addViews() {
+        LinearLayout linearLayout = new LinearLayout(getContext());
+        linearLayout.setOrientation(LinearLayout.VERTICAL);
+        linearLayout.addView(mDateButton);
+        linearLayout.addView(mDateTextView);
+        addAnswerView(linearLayout);
+    }
+
+    private void setDate() {
+        mDateTextView.setText(getAnswer().getDisplayText());
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/widgets/DateWidget.java]: I've decided to use the same method like in FormHierarchyActivity. We should keep this now and as I know someone is trying to solve https://github.com/opendatakit/collect/issues/424 and it should be a common functionality.

File: collect_app/src/main/java/org/odk/collect/android/widgets/TimeWidget.java [hunk 3]
```diff
@@ -133,17 +86,107 @@ public void setFocus(Context context) {
...
+        linearLayout.addView(mTimeButton);
+        linearLayout.addView(mTimeTextView);
+        addAnswerView(linearLayout);
+    }
+
+    private void setTime(int hourOfDay, int minuteOfHour) {
+        mHourOfDay = hourOfDay;
+        mMinuteOfHour = minuteOfHour;
+
+        mTimeTextView.setText(getAnswer().getDisplayText());
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/widgets/TimeWidget.java:142]: The same as above.

**Comment by lognaturel** [comment]
The no-calendar widgets look great! But is it not possible to make the change to those with no change to the calendar one? By putting the calendar in a dialog that can't scroll horizontally, that widget is unusable as you've pointed out. That's unfortunately not acceptable. The ideal would be to make no change at all to the calendar-based widget. The horizontal scroll is not good but organizations have worked around it as needed. That way we can make a single change to that widget in parallel with the change to material design.

**Comment by grzesiek2010** [comment]
@lognaturel I've added changes for calendar appearance and now this case is handled without any dialog.

File: collect_app/src/main/java/org/odk/collect/android/widgets/DateWidget.java (hunk 2 above)
Comment (lognaturel) [reply collect_app/src/main/java/org/odk/collect/android/widgets/DateWidget.java]: That issue is a little bit different, it's about having an opportunity to override the device locale. I'll file a separate issue for making sure `getDisplayText` matches the current locale.

**Pull request merged**